### PR TITLE
Add cli

### PR DESCRIPTION
**Pull request by facundovictor**
This PR includes:

1. The first prototype for a cabinet cli.
2. A tiny typo fix on the cabinet library.

### CODE REVIEW

File: cabinet.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    :rtype: List
+    """
+    try:
+        return value.split(',')
+    except:
+        raise argparse.argumenttypeerror("Multiple tags should be separated by\
+                                          comma")
+
+
+def tuple(value):
```
Comment (ivanalejandro0) [inline cabinet.py]: `tuple` is a reserved word, you should use other name for this function.

File: cabinet.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        """Print all the item names (and tags if apply) to stdout."""
+        account_id = 'my-name@my-company.com'
+        password = 'asdfasdf'
+        vault_name = 'test-vault'
+        tags = self.app.pargs.tags
+        if not tags:
+            tags = [] if not self.app.pargs.tag else self.app.pargs.tag
+        tag_tpl = " tagged with {1}" if self.app.pargs.show_tags else ''
+        if open_vault(account_id, password, vault_name):
+            all = get_all().values()
```
Comment (ivanalejandro0) [inline cabinet.py]: `all` is the name of a built-in function, you should use other name.

File: cabinet.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                  type=tuple, action='append'))
+        ]
+
+    @expose(help='Get an item from the vault.')
+    def get(self):
+        """Get an item from the vault"""
+        name = self.app.pargs.name
+        if name:
+            self.app.log.debug('Looking for item with name "{0}"'.format(name))
+            account_id = 'my-name@my-company.com'
```
Comment (ivanalejandro0) [inline cabinet.py]: either now that we are hardcoding credentials, or later that we may read some information from a config file and/or asking the user for it... we shouldn't repeat this all over the code.
I'd factor out all the credential stuff... maybe move it along `open_vault`?

File: cabinet.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        key, value = value.split(',')
+        return key, value
+    except:
+        raise argparse.argumenttypeerror("coordinates must be x,y,z")
+
+
+class CabinetController(CementBaseController):
+    class Meta:
+        label = 'base'
+        description = "Cabinet's cli client for managing valuts."
```
Comment (ivanalejandro0) [inline cabinet.py]: typo `s/valuts/vaults/`

File: cabinet/cabinet.py [hunk 5]
```diff
@@ -75,7 +75,7 @@ def add(self, item):
         content = item['content']
         tags = item.get('tags')
 
-        self.add_item(name, content, tags)
+        self.add_new(name, content, tags)
```
Comment (ivanalejandro0) [inline cabinet/cabinet.py:78]: good catch!

**Review by ivanalejandro0 — CHANGES_REQUESTED** [review]
I don't like very much the `global` approach of sharing the vault settings. I'd prefer to have a class that contains all the logic and we can instantiate once for later use on any command that the user calls.
That being said, I'd say that is not blocker anyway. Is better to have a working cli now than somehow improved one day :)
Good job!

File: cabinet.py (hunk 1 above)
Comment (facundovictor) [reply cabinet.py]: You're right, changed!

File: cabinet.py (hunk 4 above)
Comment (facundovictor) [reply cabinet.py]: Good catch!

File: cabinet.py (hunk 2 above)
Comment (facundovictor) [reply cabinet.py]: You're right, changed.

File: cabinet.py (hunk 3 above)
Comment (facundovictor) [reply cabinet.py]: Moved it inside a new class called `CabinetWrapper`.

**Comment by facundovictor** [comment]
I agree on removing all global vars. For now, I've moved all to a class named `CabinetWrapper`. The idea is to have a small class that we may remove in the future by moving all the useful features to the cabinet library (if apply), some features like filtering could be required by other cabinet clients. Also, the cabinet command line should be moved to it's own repo in the future.
Glad to collaborate! =)

File: cabinet/cabinet.py (hunk 5 above)
Comment (facundovictor) [reply cabinet/cabinet.py:78]: Thanks! I think github detects it as a requried change =P

**Review by ivanalejandro0 — APPROVED** [review]

**Pull request merged**